tpe/lib/src/Link.cc

```diff
@@ -42,3 +42,44 @@ Entity &Link::AddCollision()
   this->ChildrenChanged();
   return *it->second.get();
 }
+
+//////////////////////////////////////////////////
+void Link::SetLinearVelocity(const math::Vector3d _velocity)
```

Comment:
let's pass by reference. I noticed that it's the same in Model.hh but having `const &` is more efficient.

Same for `SetAngularVelocity`

Reply:
Weirdly, I've read that it isn't always the case that a const reference is more efficient for small POD statically sized data structures because of cache behaviors. Copying the three `double` values in the `Vector3d` into the stack frame may have less overhead than repeatedly dereferencing the `const Vector3d&` which is allocated elsewhere, perhaps in the heap.

That's not say we shouldn't use const-reference here. In the absence of profiling, it's fair to assume that `const T&` will be more efficient than `const T` for non-primitive types. I just thought it's interesting that the common wisdom doesn't always hold.